<?xml version="1.0" ?>
<launch>

	<arg name="mode" default="sim"/>
	<arg name="namespace" default="hugin_0"/>
    <arg name="debug" default="0"/>
	<arg name="dataset" default="overnight_2020"/>
    <arg name="app" default="basic"/>
    <arg name="manual_control" default="true"/>

    <arg name="max_throttle" default="2.0"/>

	<group ns="$(arg namespace)">
		<group ns="sim">
			<group unless="$(arg manual_control)">

				<node type="w2w_mission_planner.py" pkg="basic_navigation" name="mission_planner" output="screen">
					<param name="path_planner_as"  value="path_planner" />
					<param name="path_topic"  value="/$(arg namespace)/waypoints" />
					<param name="map_frame"  value="map" />
					<param name="wp_topic"  value="/navigation/lc_wp" />
					<param name="relocalize_topic"  value="/$(arg namespace)/pause_planner"/>
			        <param name="synch_topic" value="/$(arg app)/synch" />
			        <param name="mission_finished_top" value="/$(arg mode)/survey_finished" />
				</node>

				<node type="w2w_planner.py" pkg="basic_navigation" name="path_planner" output="screen">
					<param name="goal_tolerance"  value="5.0" />
					<param name="max_throttle"  value="$(arg max_throttle)" />
					<param name="max_thrust"  value="0.3" />
					<param name="odom_frame"  value="$(arg namespace)/odom" />
					<param name="base_frame"  value="$(arg namespace)/base_link" />
					<param name="map_frame"  value="map" />
					<param name="throttle_cmd" value="/$(arg namespace)/sim/throttle_cmd" />
					<param name="thruster_cmd" value="/$(arg namespace)/sim/thruster_cmd" />
					<param name="inclination_cmd" value="/$(arg namespace)/sim/inclination_cmd" />
					<param name="path_planner_as"  value="path_planner" />
				</node>

				<!-- <node type="backseat_pf_driver.py" pkg="backseat_driver" name="backseat_driver" output="screen">
					<param name="path_topic"  value="/$(arg namespace)/waypoints" />
					<param name="map_frame"  value="map" />
					<param name="base_frame"  value="$(arg namespace)/base_link" />
					<param name="odom_frame"  value="$(arg namespace)/odom" />
					<param name="average_pose_topic"  value="/$(arg namespace)/$(arg app)/avg_pose" />
					<param name="cov_threshold"  value="150." />
					<param name="wp_topic"  value="/navigation/lc_wp" />
					<param name="goal_tolerance"  value="5.0" />
					<param name="enable_pf_update_topic"  value="/pf_sim/enable_pf_mbes"/>
					<param name="auv_pf_launch_file"  value="$(find auv_particle_filter)/launch/auv_pf.launch"/>
					<param name="mission_launch_file"  value="$(find basic_navigation)/launch/active_pf_sim.launch"/>
					<param name="sim_path_topic"  value="/pf_sim/waypoints"/>
					<param name="relocalize_topic"  value="/$(arg namespace)/pause_planner"/>
				</node> -->
			</group>

			<group if="$(arg manual_control)">
				<node pkg="basic_navigation" type="auv_sim_teleop.py" name="auv_sim_teleop" output="screen">
					<param name="throttle_cmd" value="/$(arg namespace)/sim/throttle_cmd" />
					<param name="thruster_cmd" value="/$(arg namespace)/sim/thruster_cmd" />
					<param name="inclination_cmd" value="/$(arg namespace)/sim/inclination_cmd" />
				</node>
			</group>
		</group>
    </group>

</launch>

